src/language/pattern-types.md

```diff
@@ -0,0 +1,333 @@
+---
+title: Pattern types
+description: Pattern type reference in Dart.
+---
+
+## Cast
+
+`foo as String`
+
+Cast patterns let you insert a [type cast][] as a pattern is
+[destructuring][destructure] values:
```

Comment:
I think the original prose is deliberate, but maybe it would read better as:

> A cast pattern lets you insert a [type cast][] in the middle of destructuring before passing the value to another subpattern.

?

Reply:
Oh, yes the original meant "Cast patterns let you insert a type cast _while_ a pattern is destructuring some values", but I'll change it to Bob's suggestion